Assets/Pedestrian.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+public class Pedestrian : MonoBehaviour
+{
+    public float SpeedMagnitude = 1f;
+    public Vector3 Direction;
```

Comment:
BTW, about the Direction, you can use also the pedestrian.transform.forward to get the direction it is heading 😃.